### ISSUE TITLE
LogicalExpression tree shaking optimisation breaks with comment

### ISSUE THREAD

**Issue by tanhauhau** [rollup/rollup#3761]
- Rollup Version: 2.26.9
- Operating System (or Browser): chrome
- Node Version (if applicable):
- Link to reproduction (IMPORTANT, read below):

```js
function foo() {
	return true && // comment
		1;
}

function bar() {
	return false || // comment
		1;
}


function baz() {
	return null ?? // comment
		1;
}
	
console.log(foo());
console.log(bar());
console.log(baz());
```

[Rollup REPL](https://rollupjs.org/repl/?version=2.26.9&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmZ1bmN0aW9uJTIwZm9vKCklMjAlN0IlNUNuJTVDdHJldHVybiUyMHRydWUlMjAlMjYlMjYlMjAlMkYlMkYlMjBjb21tZW50JTVDbiU1Q3QlNUN0MSUzQiU1Q24lN0QlNUNuJTVDbmZ1bmN0aW9uJTIwYmFyKCklMjAlN0IlNUNuJTVDdHJldHVybiUyMGZhbHNlJTIwJTdDJTdDJTIwJTJGJTJGJTIwY29tbWVudCU1Q24lNUN0JTVDdDElM0IlNUNuJTdEJTVDbiU1Q24lNUNuZnVuY3Rpb24lMjBiYXooKSUyMCU3QiU1Q24lNUN0cmV0dXJuJTIwbnVsbCUyMCUzRiUzRiUyMCUyRiUyRiUyMGNvbW1lbnQlNUNuJTVDdCU1Q3QxJTNCJTVDbiU3RCU1Q24lNUN0JTVDbmNvbnNvbGUubG9nKGZvbygpKSUzQiU1Q25jb25zb2xlLmxvZyhiYXIoKSklM0IlNUNuY29uc29sZS5sb2coYmF6KCkpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==)

### Expected Behavior

should print out '1' for all cases

### Actual Behavior

prints out `undefined`

Rollup outputs:

```js
function foo() {
	return  // comment		1;
}

function bar() {
	return  // comment		1;
}


function baz() {
	return  // comment		1;
}
	
console.log(foo());
console.log(bar());
console.log(baz());
```

**Comment by tanhauhau** [comment]
related: https://github.com/rollup/rollup/pull/3734

**Comment by lukastaegert** [comment]
Thanks for spotting. So it is actually the original ASI PR #3035 that broke this, I will have a look how to fix this. So I believe the solution must be to either remove line comments completely or turn them into block comments in such situations as they always imply a line-break. I'm leaning towards removing them for now as the simple solution.

**Comment by lukastaegert** [comment]
Fix at #3762
